logging/logrus/server_interceptors.go

```diff
@@ -26,20 +26,24 @@ func UnaryServerInterceptor(entry *logrus.Entry, opts ...Option) grpc.UnaryServe
 		newCtx := newLoggerForCall(ctx, entry, info.FullMethod)
 		startTime := time.Now()
 		resp, err := handler(newCtx, req)
-		code := o.codeFunc(err)
-		level := o.levelFunc(code)
-		durField, durVal := o.durationFunc(time.Now().Sub(startTime))
-		fields := logrus.Fields{
-			"grpc.code": code.String(),
-			durField:    durVal,
-		}
-		if err != nil {
-			fields[logrus.ErrorKey] = err
+
+		if o.shouldLog(info.FullMethod, err) {
```

Comment:
I think if you reverse the condition so that the shorter branch is indented the code will be more readable.